query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -395,18 +405,28 @@ async fn group_by_aggregate(
         .collect())
 }
 
-/// Find the indexes of the relation records to traverse a set of records faster when coercing JSON values
-fn get_relation_selection_indexes<'a>(
-    selections: Vec<&'a RelationSelection>,
-    field_names: &[String],
-) -> Vec<(usize, &'a RelationSelection)> {
-    let mut output: Vec<(usize, &RelationSelection)> = Vec::new();
-
-    for (idx, field_name) in field_names.iter().enumerate() {
-        if let Some(rs) = selections.iter().find(|rq| rq.field.name() == *field_name) {
-            output.push((idx, rs));
-        }
-    }
-
-    output
+/// Find the indexes of the relation records and the virtual selection objects to traverse a set of
+/// records faster when coercing JSON values.
+fn get_selection_indexes<'a>(
```

Comment:
Um, so this function evolved from another intermediate one which doesn't exist anymore and then eventually replaced `get_relation_selection_indexes` as two code paths got refactored into one, I didn't intentionally rewrite it in functional style 😂 I don't mind either way, but let me know if you'd like to keep the previous simple imperative code and just add an `else if` inside the loop.

Reply:
It doesn't matter that much. If you feel like so, go ahead. If you don't, that's fine. Code's readable + it's a small function.